### PR TITLE
refactor: decouple node resolution from deno_core

### PR DESCRIPTION
**Pull request by dsherret**

### CODE REVIEW

File: ext/node_resolver/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+[package]
+name = "node_resolver"
```
Comment (bartlomieju) [inline ext/node_resolver/Cargo.toml:4]: Make sure to claim this crate

Comment (bartlomieju) [reply ext/node_resolver/Cargo.toml:4]: Looks like it's already claimed by us (https://crates.io/crates/node_resolver). Bump version do 0.2 then?

File: ext/node_resolver/Cargo.toml
Comment (bartlomieju) [inline ext/node_resolver/Cargo.toml:1]: A bit strange that it's an `ext/` if it's not implementing `deno_core` `extension`, but I guess not a big deal

File: ext/node_resolver/path.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+pub(crate) fn to_file_specifier(path: &Path) -> Url {
+  match Url::from_file_path(path) {
+    Ok(url) => url,
+    Err(_) => panic!("Invalid path: {}", path.display()),
+  }
+}
+
+// todo(dsherret): we have the below code also in deno_core and it
+// would be good to somehow re-use it in both places (we don't want
+// to create a dependency on deno_core here)
```
Comment (bartlomieju) [inline ext/node_resolver/path.rs:54]: Create a new crate for this? Not that this code changes often

**Review by bartlomieju — APPROVED** [review]
LGTM

File: ext/node_resolver/Cargo.toml (hunk 1 above)
Comment (dsherret) [reply ext/node_resolver/Cargo.toml:4]: It will be bumped to 0.2 on the next release.

File: ext/node_resolver/Cargo.toml
Comment (dsherret) [reply ext/node_resolver/Cargo.toml:1]: Yeah, I thought the same thing, but I didn't want to create a new top level folder 😅

Comment (dsherret) [reply ext/node_resolver/Cargo.toml:1]: I'm just going to keep it here for now. We're going to move it back to that node_resolver repo soon.

**Pull request merged**